sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/ByteArrayContent.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.implementation;
+
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Flux;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * A {@link BinaryDataContent} implementation which is backed by a {@code byte[]}.
+ */
+public class ByteArrayContent extends BinaryDataContent {
+    private final byte[] content;
+
+    public ByteArrayContent(byte[] content) {
+        if (content == null || content.length == 0) {
+            this.content = ZERO_BYTE_ARRAY;
+            return;
+        }
+        this.content = Arrays.copyOf(content, content.length);
+    }
+
+    @Override
+    public Long getLength() {
+        return (long) this.content.length;
+    }
+
+    @Override
+    public String toString() {
+        return new String(content, StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public byte[] toBytes() {
+        return Arrays.copyOf(content, content.length);
```

Comment:
Is this necessary? We've already copied the array once in the ctor, do we need to do it twice in this class?

Reply:
This is maintaining previous functionality of `BinaryData`, this is also preventing the possibility of the underlying data being manipulated and changing all copies of `toBytes`.

Reply:
As discussed, we will not support making copies of byte arrays going forward.